pkg/datapath/iptables/reconciler.go

```diff
@@ -143,15 +144,32 @@ stop:
 			}
 			state.localNodeInfo = localNodeInfo
 			stateChanged = true
-		case proxyInfo, ok := <-params.proxies:
+		case newProxyInfo, ok := <-params.proxies:
 			if !ok {
 				break stop
 			}
-			if state.proxies.Has(proxyInfo) {
+			if info, ok := state.proxies[newProxyInfo.name]; ok && info == newProxyInfo {
 				continue
 			}
-			state.proxies.Insert(proxyInfo)
-			stateChanged = true
+
+			// if existing, previous rules related to the previous entry for the same proxy name
+			// will be deleted by the manager (see Manager.addProxyRules)
+			state.proxies[newProxyInfo.name] = newProxyInfo
+
+			if !firstInit {
+				// first init not yet completed, proxy rules will be updated as part of that
+				stateChanged = true
+				continue
+			}
+
+			if err := updateProxyRules(newProxyInfo.port, newProxyInfo.isLocalOnly, newProxyInfo.name); err != nil {
```

Comment:
I'm not too familiar with how we do iptables reconciliation currently, but previous commits mention that the ordering is important here.  Is it ok to have proxy specific rules be installed on their own?

Reply:
ok, thanks for the explanation.